### DISCUSSION TITLE
How can I save the webpage created by `mitweb`

### DISCUSSION THREAD

**Discussion by sepsky** [mitmproxy/mitmproxy#5993] [Q&A]
Hi,

I'm using mitmproxy, and one of my objectives is to create a beautified logs which can be sent elsewhere to my colleague in the QA team. 

 I tried saving the session (using `-w {PATH}`), but the saved file is not really readable by them. It includes a lot of unreadable characters, as well as some logs on certificates... 
 
 I then tried `mitweb`, and the result displayed on the webpage is perfect. But when I saved it (either by passing the same `-w {PATH} or using the save function within the webpage), I get the same output that I had received earlier. 

Is there a way to save the entire webpage, so I can zip it and send it to my colleague? I'm new to mitmproxy, so any / all help is greatly appreciated.

Thanks a lot. 

Sep

**Comment by Prinzhorn** [comment]
It's not possible because it's not a static website. The React components communicate with the mitmweb instance to dynamically fetch and render the data.

One option would be to use the flows you've saved via `-w` and transform them into static websites or reports that suit your need. On a low level here's how you can read said file and access the traffic data: https://github.com/mitmproxy/mitmproxy/blob/main/examples/addons/io-read-saved-flows.py

You could also host `mitmweb` somewhere that's accessible to your colleague. You could use `--rfile` and `--no-server` to launch a readonly version that doesn't have any proxy functionality.
